src/v/storage/segment_index.h

```diff
@@ -29,6 +29,71 @@
 
 namespace storage {
 
+// clang-format off
```

Comment:
this comment is amazing. should this be surfaced in docs? cc: @Feediver1

Reply:
https://github.com/redpanda-data/documentation-private/issues/1977